cmd/user.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"github.com/cloudflare/cfssl/log"
+	"github.com/k0sproject/k0s/pkg/certificate"
+	"github.com/k0sproject/k0s/pkg/constant"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"html/template"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+func init() {
+	userCreateCmd.Flags().StringVar(&groups, "groups", "", "Specify groups")
+
+	userCmd.AddCommand(userCreateCmd)
+}
+
+var (
+	groups string
+
+	userKubeconfigTemplate = template.Must(template.New("kubeconfig").Parse(`
+apiVersion: v1
+clusters:
+- cluster:
+    server: {{.JoinURL}}
+    certificate-authority-data: {{.CACert}}
+  name: k0s
+contexts:
+- context:
+    cluster: k0s
+    user: {{.User}}
+  name: k0s
+current-context: k0s
+kind: Config
+preferences: {}
+users:
+- name: {{.User}}
+  user:
+	client-certificate-data: {{.ClientCert}}
+	client-key-data: {{.ClientKey}}
+`))
+
+	// userCmd creates new certs and kubeConfig for a user
+	userCmd = &cobra.Command{
+		Use:   "user",
+		Short: "Manage users",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+	}
+
+	userCreateCmd = &cobra.Command{
+		Use:   "create [username]",
+		Short: "Create a signed certificate for a user.",
+		Long: `Create a signed certificate and public key for a given user (and optionally user groups). <br>
+		*NOTE*: A certificate once signed cannot be revoked for a particular user.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Disable logrus and cfssl logging for user commands to avoid printing debug info to stdout
+			logrus.SetLevel(logrus.FatalLevel)
+			log.Level = log.LevelFatal
+
+			if len(args) == 0 {
+				return errors.New("Username is mandatory")
+			}
+			var username = args[0]
+			clusterConfig, err := ConfigFromYaml(cfgFile)
+			if err != nil {
+				return err
+			}
+
+			caCertPath, caCertKey := path.Join(constant.CertRootDir, "ca.crt"), path.Join(constant.CertRootDir, "ca.key")
```

Comment:
When the control plane isn't started yet, this will result in an error. I'd like to see a more meaningful error, same as in https://github.com/k0sproject/k0s/blob/main/cmd/token.go#L129-L132:
```suggestion
caCert, err := ioutil.ReadFile(path.Join(constant.CertRootDir, "ca.crt"))
if err != nil {
	return "", errors.Wrapf(err, "failed to read cluster ca certificate, is the control plane initialized on this node?")
}
caCertPath, caCertKey := path.Join(constant.CertRootDir, "ca.crt"), path.Join(constant.CertRootDir, "ca.key")
```